pages/__admin/messages/index.tsx

```diff
@@ -0,0 +1,81 @@
+import AdminMessageItem from "feature/admin-messages-list/AdminMessageItem"
+import ChannelMessageFunction from "databases/functions/ChannelMessages"
+import getConfig from "next/config"
+import init from 'nextjs-basic-auth'
+import useToastHook from "@hooks/toast/useToastHook"
+import { AdminChannelMessage } from "types/Feature/AdminMessagesList/AdminMessageItem"
+import { GetServerSidePropsContext } from "next"
+import { getAllAdminMessages } from "@services/message/adminMessageService"
+import { useRouter } from "next/router"
+import { useState } from "react"
+
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+    const {serverRuntimeConfig} = getConfig()
+    const {query} = context
+
+    const users = [
+        {user: serverRuntimeConfig.BETTER_BACKDOOR_ADMIN_USER, password: serverRuntimeConfig.BETTER_BACKDOOR_PASSWORD},
+    ]
+
+    const basicAuthCheck = init({
+        users
+    })
+
+    await basicAuthCheck(context.req, context.res)
+
+    const offset = query?.offset ? parseInt(query?.offset as string) : 0
+    const channelMessages = await ChannelMessageFunction.adminGetChannelMessage(10, offset)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The offset value is being parsed from the query string without any validation. This could lead to unexpected behavior if an invalid value is provided. Consider adding some error handling here to ensure that the offset is a valid number.

```diff
-    const offset = query?.offset ? parseInt(query?.offset as string) : 0
+    let offset = 0;
+    if (query?.offset) {
+        offset = parseInt(query?.offset as string);
+        if (isNaN(offset)) {
+            throw new Error('Invalid offset value');
+        }
+    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->